cellprofiler/modules/loadimages.py

```diff
@@ -3367,9 +3378,61 @@ def provide_image(self, image_set):
 
     def __provide_volume(self):
         pathname = url2pathname(self.get_url())
-
         if self.is_numpy_file():
             data = np.load(pathname)
+        elif self.is_omero3d_path():
+            url = self.get_url()
+            parsed_url = urlparse.urlparse(url)
+            query_params = urlparse.parse_qs(parsed_url.query)
+            zmin = int(query_params['zmin'][0])
+            zmax = int(query_params['zmax'][0])
+            width = int(query_params['width'][0])
+            height = int(query_params['height'][0])
+            stack = numpy.ndarray((zmax - zmin + 1, width, height), 'uint16')
```

Comment:
I think that the pixel types can be easily handled by `PIL` and `numpy` on their own, e.g.:
```
In [28]: response = requests.get('https://demo.glencoesoftware.com/tile/155885/0/0/0?format=tif&bsession=2482a165-695e-4
    ...: 335-bff5-220118a79dfe&name=image155885_z0_c0_t0.tif')

In [29]: ibytes = BytesIO(response.content)

In [30]: image = Image.open(ibytes)

In [31]: image
Out[31]: <PIL.TiffImagePlugin.TiffImageFile image mode=L size=640x640 at 0x7F3C74E7F1D0>

In [32]: stack = []

In [33]: stack.append(numpy.array(image))

In [34]: stack.append(numpy.array(image))

In [35]: stack.append(numpy.array(image))

In [36]: stack.append(numpy.array(image))

In [37]: stack.append(numpy.array(image))

In [38]: array3d = numpy.dstack(stack)

In [39]: array3d.shape
Out[39]: (640, 640, 5)

In [40]: array3d.dtype
Out[40]: dtype('uint8')
```

Reply:
Yep, documentation on this style (available since PIL 1.1.6) here:

 * http://effbot.org/zone/pil-changes-116.htm

Just probably have to be careful about the array shape since CellProfiler is probably expecting `(z, y, x)`.

Reply:
Interestingly this workflow is already implemented few lines above these changes: https://github.com/glencoesoftware/CellProfiler/blob/31eabb6bb13e7d0a66b5ae10449ff45838797303/cellprofiler/modules/loadimages.py#L3365

From the code it looks like the expected order is `(x, y, z)` so not sure how the `3d-omero-ms` implementation can provide a correct result, unless I'm missing something of course.

Reply:
https://github.com/glencoesoftware/CellProfiler/blob/31eabb6bb13e7d0a66b5ae10449ff45838797303/cellprofiler/modules/loadimages.py#L3384

dstack rearranges it. Isn't this z, x, y?:
`In [1]: import numpy

In [2]: stack = []

In [3]: stack.append([[1, 2],[3,4]])

In [4]: stack.append([[1, 2],[3,4]])

In [5]: stack
Out[5]: [[[1, 2], [3, 4]], [[1, 2], [3, 4]]]

In [6]: img = numpy.dstack(stack)

In [7]: img
Out[7]: 
array([[[1, 1],
        [2, 2]],

       [[3, 3],
        [4, 4]]])`

Reply:
What's the `shape` of your `img` after `.dstack`?

Reply:
Never test on square images ;) 
numpy array with dimension order z, y, x outputs the correct mask dimensions.